sdks/python/build-requirements.txt

```diff
@@ -0,0 +1,20 @@
+#
+#    Licensed to the Apache Software Foundation (ASF) under one or more
+#    contributor license agreements.  See the NOTICE file distributed with
+#    this work for additional information regarding copyright ownership.
+#    The ASF licenses this file to You under the Apache License, Version 2.0
+#    (the "License"); you may not use this file except in compliance with
+#    the License.  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# TODO(BEAM-5414): latest grpcio-tools incompatible with latest protobuf 3.6.1.
+grpcio-tools>=1.3.5,<=1.14.2
+mypy-protobuf==1.12
+future==0.16.0
```

Comment:
> This file needs a description. Which dependencies should be included here?

Good point.  These are the requirements for building an sdist for beam.  They happen to all be dependencies of `gen_protos.py`, but conceptually this file could contain other build deps like cython.  I  chose to omit that since it's a bit more complicated since  it's only used on linux and only for certain tests.

> Why is future listed here? It's already in setup.py. 

`gen_protos.py` uses it to futurize the results of `protoc`.  The  packages in `install_requires` are not available at the time that `gen_protos` runs. 

> Why is mypy-protobuf here? Only the mypy command from setup.py seems to need it, so it could be listed under REQUIRED_TEST_PACKAGES there.

Also required for `gen_protos`

Reply:
> Why is mypy-protobuf here?

To clarify, `mypy-protobuf` is required at the time that `gen_protos` runs (it's used by `protoc`), but it's not technically required by anything other than `py37-mypy` job.  However, we'll want to include the .pyi files in sdist so that end users can get their benefits in PyCharm, mypy, etc.  Which reminds me that I need to update setup.py to include pyi files in `package_data`!